Switch to bright colors on Windows Powershell

agathver opened the pull request:
Fixes #9118

Uses [bright variant](https://www.npmjs.com/package/cli-color#bright-variants) of colors from cli-color.

CMD and windows Powershell, need bright colors for propercontrast difference for comfortable reading.

agathver commented:
@abernix Regarding 03aa6d2, what if options.metaColor is passed a bright color altogether ? I currently dont see this in code, but if in future some tool is added which does pass a bright color ?

abernix commented on packages/logging/logging.js:
```diff
@@ -55,7 +55,7 @@ var META_COLOR = 'blue';
 var platform = process.platform;
 var platformColor = function (color) {
-  if (platform === 'win32') {
+  if (platform === 'win32' && !color.endsWith('Bright')) {
```
This is might be a good idea since the `logging` package could be used by outside packages and apps though I'd originally avoided it because it cut the ops per second in half over the original implementation.  It's possible I was being overzealous with optimizations, since even with this change `platformColor` still achieves 15MM ops/sec, but this is a logging function and I didn't want to introduce too much additional friction.

agathver replied:
15M ops/sec is still huge!

abernix commented:
I've dismissed my own review as it appears that there are some additional considerations to he had, and I'm mid-workflow at the moment.  Namely:

* The `logging` package is also imported [on the client](https://github.com/meteor/meteor/blob/devel/packages/logging/package.js#L21), so we can't assume `process.platform` is available.
* Since `String.prototype.endsWith` is an ECMAScript 2015 feature, it won't be available on some older clients and this package doesn't use `ecmascript`. I think it's fine to add `ecmascript` though, and it is available natively on Node.js v4+.

agathver commented:
@abernix 

> The logging package is also imported on the client, so we can't assume process.platform is available.

Fixed it

> Since String.prototype.endsWith is an ECMAScript 2015 feature, it won't be available on some older clients and this package doesn't use ecmascript. I think it's fine to add ecmascript though, and it is available natively on Node.js v4+.

Switched to String.prototype.slice, which is from ES-3

benjamn commented on packages/logging/logging.js:
```diff
@@ -49,6 +49,18 @@ var LEVEL_COLORS = {
+// switch to bright variants. While still capable of millions of
+// operations per second, the benchmark showed a 25%+ increase in
+// ops per second (on Node 8) by caching "process.platform".
+var isWin32 = process && process.platform === 'win32';
```
Unfortunately, just referring to `process` when it isn't defined will throw a `ReferenceError`, so this needs to be
```js
var isWin32 = typeof process === "object" && process.platform === "win32";
```

benjamn commented:
Thanks for working on this @agathver (and @abernix)!
